File: Digito Verificador/Modulo 10 Padrao/Teste - Arquivo 10/GerarListaMatriculasMod10.pas
program GeradorMatriculaDV;

{$mode objfpc}{$H+}

uses
  SysUtils;

function FormataMatricula(MATRICULA: Integer): string;
begin
  FormataMatricula := Format('%.6d', [MATRICULA]);
end;

function Mod10(const MATRICULA: string): Integer;
var
  MATRIC_INT: array[1..6] of Integer;
  SOMA: Integer;
  I: Integer;
begin
  for I := 1 to 6 do
    MATRIC_INT[I] := StrToInt(MATRICULA[I]);
  SOMA := 0;
  for I := 1 to 6 do
  begin
    if (I mod 2 = 0) then
    begin
      if MATRIC_INT[I] * 2 > 9 then
        SOMA := SOMA + ((MATRIC_INT[I] * 2) div 10) + ((MATRIC_INT[I] * 2) mod 10)
      else
        SOMA := SOMA + (MATRIC_INT[I] * 2);
    end
    else
      SOMA := SOMA + MATRIC_INT[I];
  end;
  Mod10 := (10 - (SOMA mod 10)) mod 10;
end;

var
  MATRICULA: string;
  I: Integer;
  DV: Integer;
  ARQUIVO: Text;

begin
  Assign(ARQUIVO, 'numeros.txt');
  Rewrite(ARQUIVO);

  Writeln('GERADOR DE NUMEROS DE MATRICULA DE 000.000 ATE 999.999 COM "DV"');
  Writeln('Padrao: Modulo 10');
  Writeln('---------------------------------------------------------------');
  Writeln;

  if IOResult <> 0 then
  begin
    Writeln('Erro ao abrir o arquivo.');
    Writeln('Programa encerrado.');
    Exit;
  end;

  Write('Aguarde arquivo sendo gerado... ');

  for I := 0 to 999999 do
  begin
    MATRICULA := FormataMatricula(I);
    DV := Mod10(MATRICULA);
    Writeln(ARQUIVO, Copy(MATRICULA, 1, 3) + '.' + Copy(MATRICULA, 4, 3) + '-' + IntToStr(DV));
  end;

  Close(ARQUIVO);

  Writeln;
  Writeln('SUCESSO: Arquivo gerado.');
end.

File: Digito Verificador/Modulo 10 Padrao/Teste - Arquivo 10/GerarListaMatriculasMod10.cpp
#include <iostream>
#include <fstream>
#include <iomanip>
using namespace std;

string formataMatricula(int MATRICULA) {
  stringstream SS;
  SS << setfill('0') << setw(6) << MATRICULA;
  return SS.str();
}

int mod10(const string& MATRICULA) {
  int SOMA = 0;
  for (int I = 0; I < 6; I++) {
    int DIGITO = MATRICULA[I] - '0';
    if (I % 2 != 0 && DIGITO * 2 > 9)
      SOMA += DIGITO * 2 - 9;
    else if (I % 2 != 0 && DIGITO * 2 <= 9)
      SOMA += DIGITO * 2;
    else
      SOMA += DIGITO;
  }
  int DV = (SOMA % 10 == 0) ? 0 : 10 - SOMA % 10;
  return DV;
}

int main(void) {
	
  string MATRICULA;
  int I, DV;	
  ofstream ARQUIVO("numeros.txt");

  cout << "GERADOR DE NUMEROS DE MATRICULA DE 000.000 ATE 999.999 COM \"DV\"" << endl;
  cout << "Padrao: Modulo 10" << endl;
  cout << "---------------------------------------------------------------" << endl;
  cout << endl;
  
  if (not ARQUIVO) {
    cerr << "Erro ao abrir o arquivo." << endl;
    cout << "Programa encerrado." << endl;
    return 1;
  }
  
  cout << "Aguarde arquivo sendo gerado... ";
  
  for (I = 0; I <= 999999; I++) {
    MATRICULA = formataMatricula(I);
    DV = mod10(MATRICULA);
    ARQUIVO << MATRICULA.substr(0, 3) << "." << MATRICULA.substr(3) << "-" << DV << endl;
  }
  
  ARQUIVO.close();
  
  cout << endl << endl;
  cout << "SUCESSO: Arquivo gerado." << endl;
  
  cout << endl;
  return 0;
}
